Classes/Utility/GeneralUtility.php

```diff
@@ -365,6 +365,30 @@ public static function doRedirectBasedOnSettings(array $settings, array $gp, str
     self::debugMessage('No redirectPage set.');
   }
 
+  /**
+   * @param array<string, mixed> $conditionSettings
+   * @param array<string, mixed> $gp
+   */
+  public static function evaluateConditions(array $conditionSettings, array $gp): false {
+    $conditionSettings = is_array($conditionSettings) ? (array) $conditionSettings : [];
+    $conditions = $conditionSettings['conditions.'] ?? [];
+    $orConditions = [];
+    foreach ($conditions as $subIdx => $andConditions) {
+      $results = [];
+      foreach ($andConditions as $subSubIdx => $andCondition) {
```

Comment:
add this:
`
      $andConditions = is_array($andConditions) ? $andConditions : [];
`
before:
`
foreach ($andConditions as $subSubIdx => $andCondition) {
`

and remove `$subSubIdx => ` as key is not needed.